.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,58 @@
+name: CI Spike (Build with Docker)
+
+on:
+  push:
+    branches: [ govuk-paas-actions-build-with-docker ]
+
+jobs:
+  build:
+    name: Build with Docker
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Docker login
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        run: docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+        
+      - name: Build and push docker image
+        env:
+          DOCKER_IMAGE_NAME: ${{ secrets.DOCKER_IMAGE_NAME }}
+        run: | # check docker permissions and ensure correct image name 
+          docker build -t $DOCKER_IMAGE_NAME .
+          docker push $DOCKER_IMAGE_NAME
+      
+      - name: Install CF CLI
```

Comment:
I'm OK with shortening Cloud Foundry in `ENV` variable keys, but let's avoid doing it in descriptive steps, please.  No need to change now, just bear it in mind moving forward.